helm/templates/api/api-deployment.yaml

```diff
@@ -112,6 +112,8 @@ spec:
               containerPort: {{ .Values.api.http.services.core.http.port | default "18093" }}
             {{- end }}
           env:
+            - name: upgrade.mode
```

Comment:
issue: as far as I know env var containing dot character is not allowed (at least not on all systems) and the convention to provide a value coming from gravitee.yaml as an environment variable is to suffix it by gravitee as described here https://documentation.gravitee.io/am/getting-started/configuration/configure-am-gateway#environment-variables. So IMO it should be: `gravitee_upgrade_mode`

Reply:
That would be more consistent with all the other properties, but - we can't use `gravitee_upgrade_mode` without changes in gravitee-node. 
That's because the [`UpgraderConfiguration`](https://github.com/gravitee-io/gravitee-node/blob/0df61fa414b74d983c1fb8437c6976064b46f5c9/gravitee-node-services/gravitee-node-services-upgrader/src/main/java/io/gravitee/node/services/upgrader/spring/UpgraderConfiguration.java#L45) that actually handles this property, uses raw system environment & properties, not the Spring `Environment` processed by `io.gravitee.node.container.spring.env.EnvironmentPropertySourceBeanProcessor`.

Reply:
indeed, I was not aware of this choise in gravitee node.